### PR TITLE
[v5] Rename `interpret` to `createActor`

### PR DESCRIPTION
**Pull request by davidkpiano**
The `interpret(...)` function has been deprecated and renamed to `createActor(...)`:

```diff
-import { interpret } from 'xstate';
+import { createActor } from 'xstate';

-const actor = interpret(machine);
+const actor = createActor(machine);
```

### CODE REVIEW

File: packages/core/src/interpreter.ts [hunk 1]
```diff
@@ -534,6 +534,34 @@ export class Interpreter<
...
+}
+
+/**
+ * Creates a new Interpreter instance for the given machine with the provided options, if any.
+ *
+ * @deprecated Use `createActor` instead
+ * @param machine The machine to interpret
+ * @param options Interpreter options
+ */
 export function interpret<TMachine extends AnyStateMachine>(
```
Comment (Andarist) [inline packages/core/src/interpreter.ts:565]: this can probably be replaced by just `export function interpret = createActor;` (with the `@deprecated` tag)

File: packages/core/src/interpreter.ts [hunk 2]
```diff
@@ -534,6 +534,34 @@ export class Interpreter<
  * @param machine The machine to interpret
```
Comment (Andarist) [inline packages/core/src/interpreter.ts:534]: Interpreter is still mentioned in the comments here. It would be cool to also rename this file and the `Interpreter` class itself

File: packages/core/src/index.ts [hunk 3]
```diff
@@ -26,7 +31,8 @@ export {
   sendTo,
   sendParent,
   forwardTo,
-  interpret,
+  createActor,
+  interpret, // deprecated
   Interpreter,
   ActorStatus as InterpreterStatus,
```
Comment (Andarist) [inline packages/core/src/index.ts:37]: `InterpreterStatus` should be phased out (or it should just be an alias of `ActorStatus` with the `@deprecated` tag). Other packages should start depending on the `ActorStatus` itself.

File: packages/core/src/interpreter.ts [hunk 4]
```diff
@@ -534,6 +534,34 @@ export class Interpreter<
...
  * @param options Interpreter options
  */
+export function createActor<TMachine extends AnyStateMachine>(
+  machine: AreAllImplementationsAssumedToBeProvided<
+    TMachine['__TResolvedTypesMeta']
+  > extends true
+    ? TMachine
+    : MissingImplementationsError<TMachine['__TResolvedTypesMeta']>,
+  options?: InterpreterOptions<TMachine>
```
Comment (Andarist) [inline packages/core/src/interpreter.ts:543]: we probably should also rename all other types that mention the interpreter, for example: `InterpreterOptions` -> `ActorOptions`

File: packages/xstate-vue/test/UseActorCreateSimple.vue [hunk 5]
```diff
@@ -13,10 +13,10 @@ import { ActorRef } from 'xstate';
 import { defineComponent, shallowRef } from 'vue';
 
 import { useActor } from '../src/index.ts';
-import { interpret } from 'xstate/src';
+import { createActor } from 'xstate/src';
```
Comment (Andarist) [inline packages/xstate-vue/test/UseActorCreateSimple.vue:16]:
```suggestion
import { createActor } from 'xstate';
```

File: .changeset/ten-singers-rhyme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+'xstate': major
+---
+
+The `interpret(...)` function has been deprecated and renamed to `createActor(...)`:
+
+```diff
+-import { interpret } from 'xstate';
++import { createActor } from 'xstate';
```
Comment (Andarist) [inline .changeset/ten-singers-rhyme.md:9]: just a note: it might be confusing that we have `createActor` and `createMachine` but they don't actually do a similar thing whereas things that do similar things to `createMachine` don't share the same naming scheme (`createMachine`, `fromPromise`, etc)

File: packages/core/src/interpreter.ts (hunk 1 above)
Comment (davidkpiano) [reply packages/core/src/interpreter.ts:565]: Ooh smart

File: .changeset/ten-singers-rhyme.md (hunk 6 above)
Comment (davidkpiano) [reply .changeset/ten-singers-rhyme.md:9]: Yeah perhaps, but the alternatives aren't really better:

- `createMachine` to `fromMachine` = we're creating a machine from a machine...?
- `fromMachineConfig` is a bit too wordy
- `createMachineFromPromise` wordy and confusing
- `createPromise` we're not creating a promise though

**Comment by davidkpiano** [comment]
Merging this in (big PR, all comments resolved, straightforward change) - won't release until next week
